File: solutions/2026-06-14/solution.java
// Minimum Window Substring
// Difficulty: Hard
// URL: https://leetcode.com/problems/minimum-window-substring/

public int minOperations(int s, int t) {
    if (s >= t) return 0;
    Queue<Integer> queue = new LinkedList<>();
    queue.offer(s);
    int operations = 0;
    while (!queue.isEmpty()) {
        int size = queue.size();
        for (int i = 0; i < size; i++) {
            int apples = queue.poll();
            if (apples == t) return operations;
            for (int[] pair : new int[][] {{-1, 1}, {1, -1}}) {
                int newApples = apples + pair[0];
                if (newApples >= 0 && newApples <= t) queue.offer(newApples);
            }
        }
        operations++;
    }
    return -1;
}

File: solutions/2026-06-14/solution.py
# Minimum Window Substring
# Difficulty: Hard
# URL: https://leetcode.com/problems/minimum-window-substring/

def minOperations(s, t):
    if s >= t:
        return 0
    queue, operations = deque([s]), 0
    while queue:
        size = len(queue)
        for _ in range(size):
            apples = queue.popleft()
            if apples == t:
                return operations
            for eat, throw in [[-1, 1], [1, -1]]:
                new_apples = apples + eat
                if 0 <= new_apples <= t:
                    queue.append(new_apples)
        operations += 1
    return -1